dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java

```diff
@@ -274,20 +274,40 @@ private boolean problemFilter(LinkageProblem linkageProblem) {
           Iterables.concat(
               getClassHierarchy(targetJavaClass),
               Arrays.asList(targetJavaClass.getAllInterfaces()));
+
+      boolean methodWithDifferentReturnTypeFound = false;
       for (JavaClass javaClass : typesToCheck) {
         for (Method method : javaClass.getMethods()) {
-          if (method.getName().equals(methodName)
-              && method.getSignature().equals(symbol.getDescriptor())) {
-            if (!isMemberAccessibleFrom(javaClass, method, sourceClassName)) {
-              return Optional.of(
-                  new InaccessibleMemberProblem(sourceClassFile, targetClassFile, symbol));
+          if (method.getName().equals(methodName)) {
+            String signatureInTargetClass = method.getSignature();
+            String signatureLookingFor = symbol.getDescriptor();
+            if (signatureInTargetClass.equals(signatureLookingFor)) {
+              if (!isMemberAccessibleFrom(javaClass, method, sourceClassName)) {
+                return Optional.of(
+                    new InaccessibleMemberProblem(sourceClassFile, targetClassFile, symbol));
+              }
+              // The method is found and accessible. Returning no error.
+              return Optional.empty();
+            } else {
+              String argumentTypeInTarget = parseArgumentTypeParts(signatureInTargetClass);
+              String argumentTypeLookingFor = parseArgumentTypeParts(signatureLookingFor);
+              if (argumentTypeInTarget.equals(argumentTypeLookingFor)) {
+                // Not returning result yet, because there can be other supertype that has the exact
```

Comment:
"another supertype" or "other supertypes"

Reply:
Updated to another supertype.